tests/test_content.py

```diff
@@ -625,6 +630,287 @@ def test_construction_staining(self):
+        processing_datetime = datetime.datetime(2023, 6, 17, 21, 38, 14)
+        issuer_of_specimen_id = IssuerOfIdentifier("issuer id")
+        fixative = CodedConcept("fixative", "test", "test fixative")
+        embedding_medium = CodedConcept("embedding", "test", "test embedding")
+        specimen_container = CodedConcept(
+            "specimen container",
+            "test",
+            "test specimen container"
+        )
+        specimen_type = CodedConcept(
+            "specimen type",
+            "test",
+            "test specimen type"
+        )
+        dataset = Dataset()
+        dataset.SpecimenPreparationStepContentItemSequence = [
+            TextContentItem(
+                name=codes.DCM.SpecimenIdentifier,
+                value=specimen_id
+            ),
+            TextContentItem(
+                name=codes.DCM.IssuerOfSpecimenIdentifier,
+                value=issuer_of_specimen_id.LocalNamespaceEntityID
+            ),
+            DateTimeContentItem(
+                name=codes.DCM.DatetimeOfProcessing,
+                value=processing_datetime
+            ),
+            CodeContentItem(
+                name=codes.DCM.ProcessingType,
+                value=processing_type
+            ),
+            TextContentItem(
+                name=codes.DCM.ProcessingStepDescription,
+                value=processing_description
+            ),
+            CodeContentItem(
+                name=codes.SCT.SpecimenCollection,
+                value=procedure
+            ),
+            CodeContentItem(
+                name=codes.SCT.TissueFixative,
+                value=fixative
+            ),
+            CodeContentItem(
+                name=codes.SCT.TissueEmbeddingMedium,
+                value=embedding_medium
+            ),
+            CodeContentItem(
+                name=codes.SCT.SpecimenContainer,
+                value=specimen_container
+            ),
+            CodeContentItem(
+                name=codes.SCT.SpecimenType,
+                value=specimen_type
+            )
+        ]
+        dataset_reread = write_and_read_dataset(dataset)
+        instance = SpecimenPreparationStep.from_dataset(dataset_reread)
+        assert isinstance(instance, SpecimenPreparationStep)
+        assert instance.specimen_id == specimen_id
+        assert instance.processing_type == processing_type
+        assert instance.fixative == fixative
+        assert instance.embedding_medium == embedding_medium
+        assert instance.processing_description == processing_description
+        processing_procedure = instance.processing_procedure
+        assert isinstance(processing_procedure, SpecimenCollection)
+        assert processing_procedure.procedure == procedure
+        assert instance.processing_datetime == processing_datetime
+        assert (
+            instance.issuer_of_specimen_id ==
+            issuer_of_specimen_id.LocalNamespaceEntityID
+        )
```

Comment:
Add check for type

Reply:
Added a type check here. Here the type should be string, as per [TID 8001](https://dicom.nema.org/medical/dicom/current/output/chtml/part16/chapter_C.html#sect_TID_8001). Or should it be a `IssuerOfIdentifier`?

Also added a type check for the same-named property in `SpecimenDescription` (here it is a `IssuerOfIdentifier`). This might have been what you asked for, given [this](https://github.com/ImagingDataCommons/highdicom/pull/246#discussion_r1332386317) comment?

Reply:
Ah, yes I meant the second one. Sorry was getting the various similarly-named classes confused.